lib/message_handler.rb

```diff
@@ -11,43 +11,62 @@ class MessageHandler < FeatureFlag
   release_for 'Fabien Townsend'
 
   def initialize(args = {})
-    @mark_all_out = args[:mark_all_out]
+    @mark_all_out = args[:mark_all_out] # TODO - this doesnt seem to be used here.
     @request_parser = RequestParser.new
     @bot = args[:bot]
     @user_info = args[:user_info_provider]
   end
 
   def handle(event_data)
-    recipient = event_data['channel'] || event_data['user']
-    data = format_data(event_data)
-    return data[:user_message] if data[:user_message].nil?
-    returned_command = @request_parser.parse(data)
+    data = format(event_data)
 
-    unless User.profile(data[:user_id])
+    if hasnt_message?(data)
+      return data[:user_message]
+    end
+
+    if hasnt_user?(data)
       User.create(data)
     end
 
-    if !User.has_office?(data[:user_id]) && !Commands::AddOffice.add_office_request?(data)
+    recipient = get_recipient(data)
+    if hasnt_office?(data)
       @bot.send("You need to add your office. ex: \"office: london\"", recipient)
       return
     end
 
-    unless returned_command.nil?
-      response = deal_with_command(returned_command)
+    command = @request_parser.parse(data)
+    unless command.nil?
+      response = run(command)
       @bot.send(response, recipient)
     end
   end
 
   private
 
-  def deal_with_command(command)
+  def get_recipient(data)
+    data[:channel_id] || data[:user_id]
+  end
+
+  def hasnt_message?(data)
```

Comment:
Nitpicking here a bit, but I generally try to avoid any names with abbreviations/accentuation.

How about `has_no_message?` ?

Reply:
Cool beans ;)